FluentAssertions.Core/Common/TypeExtensions.cs

```diff
@@ -256,14 +260,57 @@ private static bool HasNonPrivateGetter(PropertyInfo propertyInfo)
             return (getMethod != null) && !getMethod.IsPrivate && !getMethod.IsFamily;
         }
 
-        public static MethodInfo GetMethodNamed(this Type type, string methodName)
+        public static MethodInfo GetMethod(this Type type, string methodName, IEnumerable<Type> parameterTypes)
+        {
+            return type.GetMethods(AllMembersFlag)
+                .SingleOrDefault(m => m.Name == methodName && m.GetParameters().Select(p => p.ParameterType).SequenceEqual(parameterTypes));
+        }
+
+        public static bool HasMethod(this Type type, string methodName, IEnumerable<Type> parameterTypes)
+        {
+            return type.GetMethod(methodName, parameterTypes) != null;
+        }
+
+        public static MethodInfo GetParameterlessMethod(this Type type, string methodName)
+        {
+            return type.GetMethod(methodName, Enumerable.Empty<Type>());
+        }
+
+        public static bool HasParameterlessMethod(this Type type, string methodName)
         {
-            return type.GetMethod(methodName);
+            return type.GetParameterlessMethod(methodName) != null;
         }
 
-        internal static bool IsIndexer(this PropertyInfo member)
+        public static PropertyInfo GetPropertyByName(this Type type, string propertyName)
+        {
+            return type.GetProperty(propertyName, AllMembersFlag);
+        }
+
+        public static bool HasExplicitlyImplementedProperty(this Type type, Type interfaceType, string propertyName)
+        {
+            var hasGetter = type.HasParameterlessMethod(string.Format("{0}.get_{1}", interfaceType.FullName, propertyName));
```

Comment:
I wonder if there is some cleaner way to do it, something like: getter.Parent == type (Parent doesn't exist)